src/libos/src/fs/file_ops/sendfile.rs

```diff
@@ -45,8 +45,8 @@ pub async fn do_sendfile(
 
     // If offset is not none, does not modify file offset of in_fd;
     // otherwise the file offset is adjusted to reflect the number of bytes read
-    if offset.is_none() {
-        in_inode_file.seek(SeekFrom::Current(bytes_read as i64))?;
+    if offset.is_none() && bytes_read > 0 {
```

Comment:
I think this modification is wrong.

In this implementation, if read_bytes is 0 or 1, the offset will not be changed, this does not make sense. If read N bytes, the offset should be added by N, not (N-1).

Reply:
This is not a simple offset error but the read length and the write length mismatch for the socket file. Fixed.